### ISSUE TITLE
debug_traceCall has unexpected behavior

### ISSUE THREAD

**Issue by eyooooo** [maticnetwork/bor#205]
per the geth docs `The debug_traceCall method lets you run an eth_call within the context of the given block execution using the final state of parent block as the base. `

in simpler terms, if i supply blockNumber 2 to this method, my tx will be traced as if it was index 0 of block 2, using the previous block state from block 1 as its base. however, it appears that bor is placing the traced tx at the END of the inputted block. 

Is this expected behavior? is this a bug? am i doing it wrong? can i specify an index for my tx? thanks.

**Comment by temaniarpit27** [comment]
Closing this as no further updates on this
